### PR TITLE
Adding api for using appRole method

### PR DESCRIPTION
**Pull request by VAN-QIAN**
@chengshifan Please help to review

### CODE REVIEW

**Comment by chengshifan** [comment]
Hi @VAN-QIAN 
While adding vault from UI, please check if the "token" or "role_id/secret_id" exists.
One of them must required, otherwise raise error and break the adding process

**Comment by chengshifan** [comment]
Additional thing: don't forget to format code

File: src/api/handlers/job_api.py [hunk 1]
```diff
@@ -29,16 +29,31 @@
...
+
 def delete_file(path):
     if os.path.exists(path):
         try:
             os.remove(path)
         except Exception as error:
             app.logger.warn("Failed to delete file: %s", error)
 
+
+def validateway(res):
```
Comment (iberryful) [inline src/api/handlers/job_api.py:45]: rename to: get_auth_type

File: src/api/handlers/job_api.py [hunk 2]
```diff
@@ -859,36 +890,67 @@ def post(self):
...
                                 url += '/v1/' + namespace + '/' + secret_path
                             elif version == 'v2':
                                 paths = secret_path.split('/')
-                                url += '/v1/' + namespace + '/' + paths[0] + '/data/' + '/'.join(paths[1:])
+                                url += '/v1/' + namespace + '/' + \
+                                    paths[0] + '/data/' + '/'.join(paths[1:])
+                            # choose validate way
+                            validate_res = validateway(result)
+                            if validate_res == 'token':
+                                print ('validate way is token')
```
Comment (iberryful) [inline src/api/handlers/job_api.py:913]: remove this line

Comment (VAN-QIAN) [reply src/api/handlers/job_api.py]: Solved

File: src/api/handlers/job_api.py (hunk 1 above)
Comment (VAN-QIAN) [reply src/api/handlers/job_api.py]: Solved

**Pull request merged**